internal/controller/postgrescluster/apply_test.go

```diff
@@ -33,35 +33,20 @@ import (
 	"k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/version"
 	"k8s.io/client-go/discovery"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
-	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
 func TestServerSideApply(t *testing.T) {
-	// TODO: Update tests that include envtest package to better handle
-	// running in parallel
-	// t.Parallel()
-
 	ctx := context.Background()
-	env := &envtest.Environment{}
-	config, err := env.Start()
-	assert.NilError(t, err)
-	t.Cleanup(func() { assert.Check(t, env.Stop()) })
-
-	cc, err := client.New(config, client.Options{})
-	assert.NilError(t, err)
+	env, cc := setupKubernetes(t)
+	t.Parallel()
```

Comment:
> Would this be `require.ParallelCapacity(t, 0)`?

Yeah.

> Should we use both `t.Parallel` and `require.ParallelCapacity`?

We should probably use just one, right? That would be ParallelCapacity.
 
> I don't see any other `t.Parallel` so I think this could be a 🔧

I started with `require.ParallelCapacity` then noticed it was the first import of our own packages in this file. This file wasn't even using `setupEnvTest`... to make it clear that this was straight-Kubernetes? I dunno.

I'll change it to `require.ParallelCapacity(t, 0)` unless someone can think of a reason to avoid it.